src/js/main.js

```diff
@@ -0,0 +1,107 @@
+
+let searchHotel = document.getElementById('hotel-name');
+let buttonForm = document.getElementById('button-form');
+let hotels = [];
+
+window.addEventListener('load', () => {
+    loadHotels()
+});
+
+function loadHotels() {
+    let url = 'https://if-student-api.onrender.com/api/hotels';
+    fetch(url, {
+        method: 'GET',
+    })
+        .then((resp) => resp.json())
+        .then((data) => {
+            hotels = data;
+        })
+        .catch(function (error) {
+            console.log(error);
+        });
+}
+
+buttonForm.onclick = async function (e) {
+    e.preventDefault();
+    let val = searchHotel.value.toLowerCase();
+
+    if (document.getElementById('hotels')) removeNode()
+
+    loadHotels()
+
+    if (val === '') {
+        return;
+    } else {
+        hotels = hotels.filter(
+            (el) =>
+                el.name.toLowerCase().includes(val) ||
+                el.country.toLowerCase().includes(val) ||
+                el.city.toLowerCase().includes(val),
+        );
+
+    }
```

Comment:
Look at the task and use the parameter 'search' instead of this excess logic. All the filtered data you need is being prepared on the server side, just use it.